### PR TITLE
requirements: upgrade flask and other requirements (bug 1831185)

### PR DESCRIPTION
**Pull request by zzzeid**
- Update Flask and related packages
- Update test code to reflect changes in Flask

### CODE REVIEW

**Comment by zzzeid** [comment]
> CI is red, looks like some changes to Flask internals need to be reflected in Lando.
> 
> Can you file a bug for this? I also noticed our version of Flask (via `connexion`) is quite old in Lando-UI, can we upgrade those packages also?

Yep noticed this after CI ran, though locally the test initially passed (I think this was a cache issue). Fixing this up. I do have another PR for lando-ui coming up, but it has very outdated packages and thus more changes.

File: tests/test_dockerflow.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import json
 
-from tests.utils import phab_url, trans_url
```
Comment (zzzeid) [inline tests/test_dockerflow.py:7]: `trans_url` no longer used (and is deprecated functionality).

File: tests/test_dockerflow.py [hunk 2]
```diff
@@ -23,9 +23,8 @@ def test_dockerflow_version_matches_disk_contents(client, versionfile):
 
 
 def test_heartbeat_returns_200(
-    client, db, phabdouble, request_mocker, redis_cache, jwks, treestatusdouble
```
Comment (zzzeid) [inline tests/test_dockerflow.py:26]: `db` here was causing a strange app context error during teardown. Going to dig a little as to whether this is going to cause any real problems or whether we should return the db fixture here, but the error was happening on teardown.

File: tests/test_dockerflow.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import json
 
-from tests.utils import phab_url, trans_url
+from tests.utils import phab_url
```
Comment (cgsheeh) [inline tests/test_dockerflow.py:7]: Can we file a bug for removing `trans_url` and other transplant related test code?

Comment (zzzeid) [reply tests/test_dockerflow.py:7]: Sure, most of the tansplant related test code was removed in [bug 1654646](https://bugzilla.mozilla.org/show_bug.cgi?id=1654646) but there are some stragglers left like this one. Filed [bug 1831622](https://bugzilla.mozilla.org/show_bug.cgi?id=1831622).

**Pull request merged**